homework/index.js

```diff
@@ -29,16 +33,126 @@
+        const container = createAndAppend('div', root, { class: 'container' });
+        const headerDiv = createAndAppend('div', header, { class: 'headerDiv' });
+        createAndAppend('span', headerDiv, {
+          text: 'HYF Repositories',
+          id: 'headerSpan',
+        });
+        const select = createAndAppend('select', headerDiv);
+        addOptions(select, repositories);
+        let selectedRepo = repositories[select.selectedIndex];
+        const repoDetailsDiv = createAndAppend('div', container, { class: 'repoDetailsDiv' });
+        const repoDiv = createAndAppend('div', repoDetailsDiv, { class: 'divClass' });
+        const descriptionDiv = createAndAppend('div', repoDetailsDiv, {
+          class: 'divClass',
+          id: 'descDiv',
+        });
+        const forkDiv = createAndAppend('div', repoDetailsDiv, { class: 'divClass' });
+        const updatedDiv = createAndAppend('div', repoDetailsDiv, { class: 'divClass' });
+
+        createAndAppend('span', descriptionDiv, {
+          text: 'Description:',
+          class: 'spanClass',
+        });
+        createAndAppend('span', forkDiv, {
+          text: 'Forks:',
+          class: 'spanClass',
+        });
+        createAndAppend('span', updatedDiv, {
+          text: 'Updated:',
+          class: 'spanClass',
+        });
+        createAndAppend('span', repoDiv, { id: 'repoSpan', text: 'Repository:' });
+        const descP = createAndAppend('p', descriptionDiv, {
+          id: 'descP',
+          text: selectedRepo.description,
+          class: 'pClass',
+        });
+        const forkP = createAndAppend('p', forkDiv, {
+          text: selectedRepo.forks,
+          class: 'pClass',
+        });
+        const updateP = createAndAppend('p', updatedDiv, {
+          text: selectedRepo.updated_at,
+          class: 'pClass',
+        });
+        const repoA = createAndAppend('a', repoDiv, {
+          text: selectedRepo.name,
+          href: selectedRepo.html_url,
+          target: '_blank',
+        });
+
+        addContributorsDiv(selectedRepo.contributors_url, container);
+        select.onchange = function func() {
+          selectedRepo = repositories[select.selectedIndex];
+          repoA.href = selectedRepo.html_url;
+          repoA.innerText = selectedRepo.name;
+          descP.innerText = selectedRepo.description;
+          forkP.innerText = selectedRepo.forks;
+          updateP.innerText = selectedRepo.updated_at;
+          container.removeChild(document.getElementById('rightDiv'));
+          if (!descP.innerText) {
+            document.getElementById('descDiv').style.display = 'none';
+          } else document.getElementById('descDiv').style.display = 'flex';
+
+          addContributorsDiv(selectedRepo.contributors_url, container);
+        };
+      })
+      .catch(err => {
+        const root = document.getElementById('root');
         createAndAppend('div', root, { text: err.message, class: 'alert-error' });
```

Comment:
This is the absolute minimal implementation for rendering the error. It would be nice to show the top toolbar as shown in the assignment example.

![image](https://user-images.githubusercontent.com/2788771/59962240-f8406400-94e2-11e9-9450-6ffc5dfa5f58.png)